src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -203,14 +203,16 @@ public DynamoView(DynamoViewModel dynamoViewModel)
         }
 
         /// <summary>
-        /// This method close a tab item in the right side bar based on passed extension
+        /// This method will close a tab item in the right side bar based on passed extension
         /// </summary>
         /// <param name="viewExtension">Extension to be closed</param>
         /// <returns></returns>
         internal void CloseTabItem(IViewExtension viewExtension)
         {
             string tabName = viewExtension.Name;
-            CloseTab(tabName);
+
+            TabItem tabitem = TabItems.OfType<TabItem>().SingleOrDefault(n => n.Name == tabName);
```

Comment:
Please switch to use UID here

Reply:
@QilongTang When an extension tab is closed using the "close" button, the sender object only has reference to the extension name and it does not pass the extension Uid. https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs#L258

Since we do not have a dictionary to retrieve the Uid using the name of the extension or vice-versa, this would be difficult. The extension names in the side bar are also unique in the sense that if any extension with the same name is added to the extension side bar, it would not add it. This will take care of that. https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs#L222

Even though it is definitely better to use Uid property, there is no issue using the extension names as of now.

Reply:
@reddyashish Cool, then we can stick to name for now and move on to GUID later. I just hope it is not going to be too many instances where people's work can't load due to other people's extension naming